### PR TITLE
Chrome extension support

### LINKED ISSUE #148: Chrome Extension Support
Issue: react-hook-form/react-hook-form#8624
Pull request: react-hook-form/react-hook-form#8628

## Usage
If you have chrome extension. The component won't be displayed and extension will replace the component instead.

### PR DESCRIPTION
**Pull request by jsun969**
close #148

### CODE REVIEW

**Comment by jsun969** [comment]
Please fix #149 first cuz it need to be tested in example.
I've tested in `devToolUI.tsx` (cuz storybook only need ui) but it's not a good way.

File: src/devTool.tsx [hunk 1]
```diff
@@ -25,6 +26,13 @@ export const DevTool = <T extends FieldValues>(props?: {
 }) => {
   const methods = useFormContext();
 
+  const { isExtensionEnabled } = useExportControlToExtension(
+    (props && props.control) || methods.control,
```
Comment (bluebill1049) [inline src/devTool.tsx:30]:
```suggestion
    props?.control ?? methods.control,
```

File: src/extension/useExportControlToExtension.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    } as MessageData);
+    setIsExtensionEnabled(true);
+  };
+
+  useEffect(() => {
+    window.addEventListener('message', handleInitMessage);
+    return () => window.removeEventListener('message', handleInitMessage);
+  }, []);
+
+  const toFlat = <V>(obj: object, defaultValue?: V) => {
```
Comment (bluebill1049) [inline src/extension/useExportControlToExtension.ts:36]: Could we move this into utility?

**Review by bluebill1049 — DISMISSED** [review]
LGTM, thanks for the PR~!

File: src/extension/useExportControlToExtension.ts (hunk 2 above)
Comment (jsun969) [reply src/extension/useExportControlToExtension.ts:36]: It's not utility, needs `control._names`.
But sure, I can make it as untility

Comment (bluebill1049) [reply src/extension/useExportControlToExtension.ts:36]: Got you, you can make _names as payload? no biggy, just try to move some of the mapping logic into own function.

File: src/devTool.tsx (hunk 1 above)
Comment (jsun969) [reply src/devTool.tsx:30]: Also Line 39?

Comment (bluebill1049) [reply src/devTool.tsx:30]: again no biggy, just minor I have already approved, happy to merge as it is.

**Review by bluebill1049 — APPROVED** [review]

**Pull request merged**